lib/Driver/ToolChains/CommonArgs.cpp

```diff
@@ -870,6 +870,21 @@ tools::ParsePICArgs(const ToolChain &ToolChain, const ArgList &Args) {
     PIP = true;
   }
 
+  // Pagerando and '-ffunction-sections'/'-fdata-sections' are incompatible
```

Comment:
It could make more sense to check for `-fpagerando` or whatever when consuming `-ffunction/data-sections` instead of here. Then you wouldn't need the duplicated `UseSeparateSections` etc. I'm really not sure which place makes more sense here, I can see reasons for both.